### PR TITLE
ScanCode: Improve mapping timeout errors

### PR DESCRIPTION
**Pull request by fviernau**
Relax the precondition for mapping timeout errors, so that the mapping is applied
regardless of the actual timeout value.

Also cover the mapping by the tests, which required to re-create the test assets for the output formats. 

Note: This prepares for setting `Issue.affectedPath` inside of `mapTimeoutIssues()`.

Part of #7921.

### CODE REVIEW

File: plugins/scanners/scancode/src/test/assets/scancode-output-format-2.0.0_mime-types-2.1.18.json [hunk 1]
```diff
@@ -13,25 +13,27 @@
         "--license": true,
         "--processes": "3",
         "--strip-root": true,
-        "--timeout": "300.0"
+        "--timeout": "0.09"
```
Comment (sschuberth) [inline plugins/scanners/scancode/src/test/assets/scancode-output-format-2.0.0_mime-types-2.1.18.json:16]: Any reason why you using 0.09 here but 0.07 above? Similar for the next file.

File: plugins/scanners/scancode/src/test/kotlin/ScanCodeResultParserTest.kt [hunk 2]
```diff
@@ -179,3 +178,5 @@ private fun containCopyrightsExactly(vararg copyrights: Pair<String, List<TextLo
         summary.copyrightFindings.groupBy { it.statement }.entries
             .map { (key, value) -> key to value.map { it.location } }
     }
+
+private fun getAssetFile(path: String): File = File("src/test/assets", path).absoluteFile
```
Comment (sschuberth) [inline plugins/scanners/scancode/src/test/kotlin/ScanCodeResultParserTest.kt:179]: Please just import the existing function from `test-utils`. Also plugin projects have access to it.

**Review by sschuberth — CHANGES_REQUESTED** [review]

File: plugins/scanners/scancode/src/test/assets/scancode-output-format-2.0.0_mime-types-2.1.18.json (hunk 1 above)
Comment (fviernau) [reply plugins/scanners/scancode/src/test/assets/scancode-output-format-2.0.0_mime-types-2.1.18.json:16]: I played with the values until I got exactly one issue.

File: plugins/scanners/scancode/src/test/kotlin/ScanCodeResultParserTest.kt (hunk 2 above)
Comment (fviernau) [reply plugins/scanners/scancode/src/test/kotlin/ScanCodeResultParserTest.kt:179]: The existing functions only works for `funTest`.

Comment (sschuberth) [reply plugins/scanners/scancode/src/test/kotlin/ScanCodeResultParserTest.kt:179]: It shouldn't, as the `test-utils` dependency is added for *all* test suite, not just funTests:

https://github.com/oss-review-toolkit/ort/blob/521640b4f55edfbfb357f65a12db480eb9521474/buildSrc/src/main/kotlin/ort-kotlin-conventions.gradle.kts#L62

I actually tried it quickly, and it worked for me. Just the import needs to be added.

Comment (fviernau) [reply plugins/scanners/scancode/src/test/kotlin/ScanCodeResultParserTest.kt:179]: The implementation looks like this:

```
fun getAssetFile(path: String): File = File("src/funTest/assets", path).absoluteFile
```

but the asset is under `src/test/assets`, not `src/FunTest/assets`.

Comment (sschuberth) [reply plugins/scanners/scancode/src/test/kotlin/ScanCodeResultParserTest.kt:179]: Ah, sorry, I misread your comment.

**Review by sschuberth — APPROVED** [review]

**Pull request merged**